src/proxyvisor.coffee

```diff
@@ -0,0 +1,262 @@
+
+router.put '/v1/devices/:uuid', (req, res) ->
+	uuid = req.params.uuid
+	status = req.body.status
+	is_online = req.body.is_online
+	if is_online? and !_.isBoolean(is_online)
+		res.status(400).send('is_online must be a boolean')
+		return
+	if status? and _.isEmpty(status)
+		res.status(400).send('status cannot be empty')
+		return
+	Promise.join(
+		utils.getConfig('apiKey')
+		knex('dependentDevice').select().where({ uuid })
+		(apiKey, [ device ]) ->
+			throw new Error('apikey not found') if !apiKey?
+			return res.status(404).send('Device not found') if !device?
+			resinApi.patch
+				resource: 'device'
+				id: device.deviceId
+				body: { status, is_online }
+				customOptions:
+					apikey: apiKey
+			.then ->
+				device.status = status
+				device.is_online = is_online
+				knex('dependentDevice').update(device).where({ uuid })
+			.then ->
+				res.json(device)
+	)
+	.catch (err) ->
+		console.error('Error on PUT /v1/devices:', err, err.stack)
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+tarPath = (app) ->
+	return '/tmp/' + app.commit + '.tar'
+
+router.get '/v1/assets/:appId/:commit', (req, res) ->
+	knex('dependentApp').select().where(_.pick(req.params, 'appId', 'commit'))
+	.then ([ app ]) ->
+		return res.status(404).send('Not found') if !app
+		dest = tarPath(app)
+		getAssetsPath(app.imageId)
+		.then (path) ->
+			getTarArchive(path, dest)
+		.then ->
+			res.sendFile(dest)
+	.catch (err) ->
+		console.error('Error on GET /v1/assets/:commit:', err, err.stack)
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.get '/v1/dependent-apps', (req, res) ->
+	knex('dependentApp').select()
+	.map (app) ->
+		return {
+			id: app.appId
+			commit: app.commit
+			device_type: 'edge'
+			config: app.config
+		}
+	.then (apps) ->
+		res.json(apps)
+	.catch (err) ->
+		console.error('Error on GET /v1/dependent-apps', err, err.stack)
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+getTarArchive = (path, destination) ->
+	fs.lstatAsync(path)
+	.then ->
+		execAsync("cd #{path} && tar -cvf #{destination} *")
+
+# TODO: deduplicate code from compareForUpdate in application.coffee
+exports.fetchAndSetTargetsForDependentApps = (state, fetchFn) ->
+	knex('dependentApp').select()
+	.then (localDependentApps) ->
+		# Compare to see which to fetch, and which to delete
+		remoteApps = _.mapValues state.apps, (app, appId) ->
+			conf = app.config ? {}
+			return {
+				appId: appId
+				parentAppId: app.parentApp
+				imageId: app.image
+				commit: app.commit
+				config: JSON.stringify(conf)
+			}
+		localApps = _.indexBy(localDependentApps, 'appId')
+
+		toBeDownloaded = _.filter remoteApps, (app, appId) ->
+			return  app.commit? and app.imageId? and !_.any(localApps, imageId: app.imageId)
+		toBeRemoved = _.filter localApps, (app, appId) ->
+			return app.commit? and !_.any(remoteApps, imageId: app.imageId)
+		Promise.map toBeDownloaded, (app) ->
```

Comment:
Yeah, but it's gonna be a different set of update strategies, so better leave that for a future PR